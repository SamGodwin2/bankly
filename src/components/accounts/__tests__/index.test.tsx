import * as React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import Accounts from '../';

const renderer = ShallowRenderer.createRenderer()

describe('Account component', () => {

  const mockAccounts = [
    {
      account_id: '1234',
      balance: {
        amount: {
          currency: 'GBP',
          value: 27800.10
        }
      }
    },
    {
      account_id: '5678',
      balance: {
        amount: {
          currency: 'EUR',
          value: 2000.50
        }
      }
    }
  ]

  test('should render and match the snapshot', () => {
    renderer.render(<Accounts accounts={mockAccounts}/>)
    const output = renderer.getRenderOutput()

    expect(output).toMatchSnapshot();
  });
});
