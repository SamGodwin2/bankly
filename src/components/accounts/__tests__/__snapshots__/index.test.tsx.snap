// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Account component should render and match the snapshot 1`] = `
<div
  className="sc-bcXHqe eEQRJM"
>
  <div
    className="account"
    id="1234"
  >
    <div
      className="total"
      id="total-1234"
    >
      Total 
      GBP
    </div>
    <h3
      aria-labelledby="total-1234"
    >
      <strong>
        27800.1
      </strong>
    </h3>
  </div>
  <div
    className="account"
    id="5678"
  >
    <div
      className="total"
      id="total-5678"
    >
      Total 
      EUR
    </div>
    <h3
      aria-labelledby="total-5678"
    >
      <strong>
        2000.5
      </strong>
    </h3>
  </div>
</div>
`;
